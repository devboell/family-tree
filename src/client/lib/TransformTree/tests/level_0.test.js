import { singleRoot, withPartner, withTwoPartners } from '../test_data/level_0'
import { transformTree } from '../index'

const dimensions = {
  boxWidth: 80,
  boxHeight: 80,
  marginX: 20,
  marginY: 20,
}
it('should transform singleRoot', () => {
  expect(transformTree(singleRoot, dimensions)).toMatchSnapshot()
})

it('should transform withPartner', () => {
  expect(transformTree(withPartner, dimensions)).toMatchSnapshot()
})

it('should transform withTwoPartners', () => {
  expect(transformTree(withTwoPartners, dimensions)).toMatchSnapshot()
})
