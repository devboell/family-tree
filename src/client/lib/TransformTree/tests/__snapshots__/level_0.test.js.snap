// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should transform singleRoot 1`] = `
Object {
  "level": 0,
  "partners": Array [],
  "totalHeight": 100,
  "totalWidth": 100,
  "value": "root",
}
`;

exports[`should transform withPartner 1`] = `
Object {
  "level": 0,
  "partners": Array [
    Object {
      "children": Array [],
      "level": 0,
      "totalHeight": 100,
      "totalWidth": 200,
      "value": "partner root",
    },
  ],
  "totalHeight": 100,
  "totalWidth": 200,
  "value": "root",
}
`;

exports[`should transform withTwoPartners 1`] = `
Object {
  "level": 0,
  "partners": Array [
    Object {
      "children": Array [],
      "level": 0,
      "totalHeight": 100,
      "totalWidth": 200,
      "value": "partner 1 root",
    },
    Object {
      "children": Array [],
      "level": 0,
      "totalHeight": 100,
      "totalWidth": 200,
      "value": "partner 2 root",
    },
  ],
  "totalHeight": 100,
  "totalWidth": 400,
  "value": "root",
}
`;
